### PR TITLE
k8s-infra-prow-build: add Ingress and Service for Grafana

### PR DESCRIPTION
**Pull request by xmudrii**
Follow up on #6484 based on https://kubernetes.slack.com/archives/CCK68P2Q2/p1709051458404019

/assign @ameukam 
cc @koksay

### CODE REVIEW

File: infra/gcp/terraform/k8s-infra-prow-build/prow-build/resources/monitoring/grafana/service.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: monitoring
+  name: grafana
+  labels:
+    app: grafana
+spec:
+  type: ClusterIP
```
Comment (ameukam) [inline infra/gcp/terraform/k8s-infra-prow-build/prow-build/resources/monitoring/grafana/service.yaml:9]: @xmudrii Need to be `LoadBalancer`. 
I miss this.

Comment (xmudrii) [reply infra/gcp/terraform/k8s-infra-prow-build/prow-build/resources/monitoring/grafana/service.yaml:9]: ACK, will create a PR to fix this.

Comment (xmudrii) [reply infra/gcp/terraform/k8s-infra-prow-build/prow-build/resources/monitoring/grafana/service.yaml:9]: https://github.com/kubernetes/k8s.io/pull/6488

**Pull request merged**